### PR TITLE
Updated systems engineering challenge.

### PR DESCRIPTION
**Pull request by russjones**
**Description**

Converted the challenge to markdown to make it easier to make changes and made a few tweaks to make our requirements clearer.

### CODE REVIEW

File: challenges/systems/challenge.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+* CLI should be able to query result of the job execution and fetch its output.
+
+## Level 3
+
+### Library
+
+* Worker library with methods to start/stop/query status and get the output of a job.
+* Library should be able to stream the output of a running job.
+  * Output should be from start of process execution. 
+  * Multiple concurrent clients should be supported.
```
Comment (russjones) [inline challenges/systems/challenge.md:73]: Added more details about streaming.

File: challenges/systems/challenge.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+### Level 4
+
+### Library
+
+* Worker library with methods to start/stop/query status and get the output of a job.
+* Library should be able to stream the output of a running job.
+  * Output should be from start of process execution. 
+  * Multiple concurrent clients should be supported.
+* Add resource control for CPU, Memory and Disk IO per job using cgroups.
```
Comment (russjones) [inline challenges/systems/challenge.md:94]: Made it clear we expect cgroups for resource control.

File: challenges/systems/challenge.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+# Guidance
+
+## Interview process
+
+The interview team will join the Slack channel. The team consists of the engineers who will be working with you. Ask them about the engineering culture, work and life balance, or anything else that you would like to learn about Teleport.
+
+Before writing the actual code, create a brief design document in Google Docs or markdown in Github and share with the team.
+
+This document should consist of key trade-offs, design approaches, scope and tradeoffs, security, and API.
```
Comment (russjones) [inline challenges/systems/challenge.md:137]: Made it clear what we expect in the design document.

File: challenges/systems/challenge.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+Unless specified in the requirements, pick the solution that works best for you.
+
+# Tools
+
+This task should be implemented in Go or Rust and should work on 64-bit Linux machines.
+
+# Timing
+
+You can split coding over a couple of weekdays or weekends and find time to ask questions and receive feedback.
```
Comment (russjones) [inline challenges/systems/challenge.md:230]: Removed timing hours.

**Review by klizhentas — APPROVED** [review]

**Review by knisbet — APPROVED** [review]

**Pull request merged**